ingestion/src/metadata/ingestion/source/pipeline/openlineage/metadata.py

```diff
@@ -0,0 +1,534 @@
+
+        # for some OL events name can be extracted from dataset facet but symlinks is preferred so - if present - we
+        # use it instead
+        if len(symlinks) > 0:
+            try:
+                # @todo verify if table can have multiple identifiers pointing at it
+                name_parts = symlinks[0]["name"].split(".")
+            except (KeyError, IndexError):
+                raise ValueError(
+                    "input table name cannot be retrieved from symlinks.identifiers facet."
+                )
+        else:
+            try:
+                name_parts = data["name"].split(".")
+            except KeyError:
+                raise ValueError(
+                    "input table name cannot be retrieved from name attribute."
+                )
+
+        # we take last two elements to explicitly collect schema and table names
+        # in BigQuery Open Lineage events name_parts would be list of 3 elements as first one is GCP Project ID
+        # however, concept of GCP Project ID is not represented in Open Metadata and hence - we need to skip this part
+        return TableDetails(name=name_parts[-1], schema=name_parts[-2])
+
+    def _get_table_fqn(self, table_details: TableDetails) -> Optional[str]:
+        try:
+            return self._get_table_fqn_from_om(table_details)
+        except FQNNotFoundException:
+            try:
+                schema_fqn = self._get_schema_fqn_from_om(table_details.schema)
+
+                return f"{schema_fqn}.{table_details.name}"
+            except FQNNotFoundException:
+                return None
+
+    def _get_entity_fqn_from_om(
+        self, entity_type: Union[type(Table), type(DatabaseSchema)], **kwargs
+    ) -> Optional[str]:
+        fully_qualified_name = fqn.build(
+            metadata=self.metadata, entity_type=entity_type, **kwargs
+        )
+        if fully_qualified_name:
+            entity: Union[Table, DatabaseSchema] = self.metadata.get_by_name(
+                entity=entity_type, fqn=fully_qualified_name
+            )
+            if entity:
+                result = entity.fullyQualifiedName.__root__
+                return result
+
+        return None
+
+    def _get_table_fqn_from_om(self, table_details: TableDetails) -> Optional[str]:
+        """
+        Based on partial schema and table names look for matching table object in open metadata.
+        :param schema: schema name
+        :param table: table name
+        :return: fully qualified name of a Table in Open Metadata
+        """
+        result = None
+        services = self.source_config.dbServiceNames
+        for db_service in services:
+            result = self._get_entity_fqn_from_om(
+                Table,
+                service_name=db_service,
+                table_name=table_details.name,
+                schema_name=table_details.schema,
+                database_name=None,
+            )
+            if result:
```

Comment:
@mgorsk1 you do not need to write a method to validate if the table exist in openmetadata or not, when you execute method `fqn.build` and when you are not passing the database name the function itself will make call to ES and if the table exists in openmetadata then it would return an appropriate fqn otherwise it would return None, that is how you can ensure if table exists

https://github.com/open-metadata/OpenMetadata/blob/e8c0b017ba9e4f88e53b11c8412b99d261b9f394/ingestion/src/metadata/utils/fqn.py#L183

Reply:
ok, removed this if clause